simulationdataschema/model_system_review.py

```diff
@@ -0,0 +1,865 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD.
+# See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import numpy as np
+import ase
+
+from matid import SymmetryAnalyzer, Classifier  # pylint: disable=import-error
+from matid.classification.classifications import (
+    Class0D,
+    Atom,
+    Class1D,
+    Class2D,
+    Material2D,
+    Surface,
+    Class3D,
+)  # pylint: disable=import-error
+
+from nomad import config
+from nomad.units import ureg
+from nomad.atomutils import Formula, get_normalized_wyckoff, search_aflow_prototype
+
+from nomad.datamodel.data import ArchiveSection
+
+from nomad.metainfo import Quantity, SubSection, SectionProxy, MEnum
+from nomad.datamodel.metainfo.basesections import System, GeometricSpace
+from nomad.datamodel.metainfo.annotations import ELNAnnotation
+
+
+def check_parent_and_atomic_cell(section, logger):
+    """
+    Checks if the parent of a section exists and whether it has a sub-section atomic_cell.
+    This is useful for other sub-sections under ModelSystem. It returns then the corresponding
+    AtomicCell section.
```

Comment:
I think not, because this checks that whatever the `section` is, if it has a m_parent and this m_parent has an atomic_cell.

In any case, I think I can extend this and use it independently as an utility function (passing also the sub_section I want to resolve from the parent). It is a bit patchy, but might also be useful when going into some ladder of sections/sub-sections.

Reply:
I think not. The goal of this method is to do some seesaw movement in `section` to return `sub_section`, i.e., one section up, one section down. It does not require that we are using `AtomicCell`, but could be useful in the future when dealing with this entangled sections and normalizations.

I will define `utils/utils.py` and add this as `get_sub_section_from_section_parent(section, sub_section, logger)` (or `seesaw_section_to_sub_section(section, sub_section, logger)`). Feel free to suggest better naming.

Reply:
I'm not following... I don't think you mean depth navigation, but rather an order of normalizer execution?
My suggestion was based on its current usage. Maybe it's faster if you clarify in person...

For actual navigation (during construction), I'm building something out in a separate branch.